langchain4j-azure-cosmos-nosql/src/main/java/dev/langchain4j/store/embedding/azure/cosmos/nosql/AbstractAzureCosmosDBNoSqlEmbeddingStore.java

```diff
@@ -0,0 +1,674 @@
+    protected static final String DEFAULT_FULL_TEXT_INDEX_PATH = "/text";
+    protected static final String DEFAULT_FULL_TEXT_INDEX_LANGUAGE = "en-US";
+
+    private static final Logger logger = LoggerFactory.getLogger(AbstractAzureCosmosDBNoSqlEmbeddingStore.class);
+    protected AzureCosmosDBNoSqlFilterMapper filterMapper;
+    Integer vectorQuantizationSizeInBytes;
+    Integer vectorIndexingSearchListSize;
+    List<String> vectorIndexShardKeys;
+    String fullTextIndexPath;
+    String fullTextIndexLanguage;
+    Boolean hybridOrFullTextSearchEnabled;
+    private CosmosAsyncClient cosmosClient;
+    private String databaseName;
+    private String containerName;
+    private String partitionKeyPath;
+    private Integer vectorStoreThroughput;
+    private AzureCosmosDBSearchQueryType searchQueryType;
+    private Integer vectorDimensions;
+    private String vectorIndexPath;
+    private String vectorIndexType;
+    private String vectorDataType;
+    private String vectorDistanceFunction;
+    private CosmosAsyncContainer container;
+
+    protected void initialize(
+            String endpoint,
+            AzureKeyCredential keyCredential,
+            TokenCredential tokenCredential,
+            String databaseName,
+            String containerName,
+            String partitionKeyPath,
+            Integer vectorStoreThroughput,
+            AzureCosmosDBSearchQueryType searchQueryType,
+            String vectorIndexType,
+            String vectorIndexPath,
+            String vectorDataType,
+            Integer vectorDimensions,
+            String vectorDistanceFunction,
+            Integer vectorQuantizationSizeInBytes,
+            Integer vectorIndexingSearchListSize,
+            List<String> vectorIndexShardKeys,
+            String fullTextIndexPath,
+            String fullTextIndexLanguage,
+            AzureCosmosDBNoSqlFilterMapper filterMapper) {
+        ensureNotNull(endpoint, "%s", "cosmosClient cannot be null or empty for Azure CosmosDB NoSql Embedding Store.");
+
+        if (filterMapper == null) {
+            this.filterMapper = new DefaultAzureCosmosDBNoSqlFilterMapper();
+        } else {
+            this.filterMapper = filterMapper;
+        }
+        try {
+            if (keyCredential != null) {
+                this.cosmosClient = new CosmosClientBuilder()
+                        .endpoint(endpoint)
+                        .credential(keyCredential)
+                        .userAgentSuffix(USER_AGENT)
+                        .buildAsyncClient();
+            } else {
+                this.cosmosClient = new CosmosClientBuilder()
+                        .endpoint(endpoint)
+                        .credential(tokenCredential)
+                        .userAgentSuffix(USER_AGENT)
+                        .buildAsyncClient();
+            }
+
+        } catch (Exception e) {
+            logger.error("Error creating cosmosClient: {}", e.getMessage());
```

Comment:
```suggestion
            logger.error("Error creating cosmosClient", e);
```
But does it make sense to continue without `cosmosClient`?

Reply:
Updated to throw an exception. Nice catch.